chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+            #        tsopt = ["-"]
+            #        mdopt = ["-"]
+            #        bxnd = 4
+            #    if (ndxt > 9) or (ndxt < 0):
+            #        ndxt = 0   # trap ff
+            #    if ndxm > 2:
+            #        ndxm = 0
+            # # end if fhigh > 0
+            # rx = RadioSettingValueList(tsopt, tsopt[ndxt])
+            # rset = RadioSetting("pgmscanedge/%d.tstp" % kx,
+            #                    "-- Scan %d Freq Step" % kx, rx)
+            # rset.set_apply_callback(myset_tsopt, _pses, kx, "tstp", bxnd)
+            # edges.append(rset)
+
+            # rx = RadioSettingValueList(mdopt, mdopt[ndxm])
+            # rset = RadioSetting("pgmscanedge/%d.mode" % kx,
+            #                    "-- Scan %d Mode" % kx, rx)
+            # rset.set_apply_callback(myset_mdopt, _pses, kx, "mode", bxnd)
+            # edges.append(rset)
+        # End for kx (Edges)
+
+        # -------------------
+        #   COMMON SETTINGS
+        # -------------------
+
+        # Set Mode - Func + Up/Down
+        # Set Mode - Power Save
+        # Set Mode - Dial Function
+        # Sounds - Key Beep
+        # Sounds - Beep level
+        # Display - Backlight
+        # Display - Memory Display Name
+
+        # Antenna - AM
+        options = ["Ext", "Bar"]
+        rx = RadioSettingValueList(options, options[_sets.am_ant])
+        rset = RadioSetting("settings.am_ant", "AM Antenna", rx)
+        common.append(rset)
+
+        # Antenna - FM
+        options = ["Ext", "Ear"]
+        rx = RadioSettingValueList(options, options[_sets.fm_ant])
+        rset = RadioSetting("settings.fm_ant", "FM Antenna", rx)
+        common.append(rset)
+
+        # CIV Address
+        stx = str(_sets.civ_address)[2:]    # Hex value
+        rx = RadioSettingValueString(1, 2, stx)
+        rset = RadioSetting("settings.civ_address", "CI-V Address (7E)", rx)
+        # rset.set_apply_callback(hex_val, _sets, "civ_address")
+        common.append(rset)
+
+        # CIV Baud:
+        rx = RadioSettingValueList(CIV_BAUD_RATES,
+                                   CIV_BAUD_RATES[_sets.civ_baud_rate])
+        rset = RadioSetting("settings.civ_baud_rate", "CI-V Baud Rate", rx)
+        common.append(rset)
+
+        # CIV - Transmit frequency/mode changes
+        rx = RadioSettingValueBoolean(bool(_sets.civ_transceive))
+        rset = RadioSetting("settings.civ_transceive", "CI-V Transceive", rx)
+        common.append(rset)
+
+        # Scan - Program Skip Scan
+        # Scan - Bank Link A-H
+        # Scan - Bank Link I-P
+        # Scan - Bank Link Q-Y
+        # Scan - Pause Timer
+        # Scan - Resume Timer
+        # Scan - Stop Beep
+
+        return group
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number]
+        _flag = self._memobj.flags[mem.number]
+
+        if mem.empty:
+            _flag.hide_channel = 1
+            return
+        _flag.hide_channel = 0
```

Comment:
This is a nit, but above the if, just do this:
```
_flag.hide_channel = mem.empty
```
and it'll do the right thing :)

Reply:
Fixed